src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpCodeActions.cs

```diff
@@ -50,12 +55,248 @@ public static void Main(string[] args)
+
+            this.VerifyTextContains(
+                @"
+using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        Exception ex = new ArgumentException();
+    }
+}");
+
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvokeDelegateWithConditionalAccess)]
+        public void InvokeDelegateWithConditionalAccessMultipleTimes()
+        {
+            var markup = @"
+using System;
+class C
+{
+    public event EventHandler First;
+    public event EventHandler Second;
+    void RaiseFirst()
+    {
+        var temp1 = First;
+        if (temp1 != null)
+        {
+            temp1$$(this, EventArgs.Empty);
+        }
+    }
+    void RaiseSecond()
+    {
+        var temp2 = Second;
+        if (temp2 != null)
+        {
+            temp2(this, EventArgs.Empty);
+        }
+    }
+}";
+
+            MarkupTestFile.GetSpans(markup, out var text, out ImmutableArray<TextSpan> spans);
+
+            SetUpEditor(markup);
+            this.InvokeCodeActionList();
+            this.VerifyCodeAction("Delegate invocation can be simplified.", applyFix: true, ensureExpectedItemsAreOrdered: true, blockUntilComplete: true);
+
+            Editor.PlaceCaret("temp2", 0, 0, extendSelection: false, selectBlock: false);
+            this.InvokeCodeActionList();
+            this.VerifyCodeAction("Delegate invocation can be simplified.", applyFix: true, ensureExpectedItemsAreOrdered: true, blockUntilComplete: true);
+            this.VerifyTextContains("First?.");
+            this.VerifyTextContains("Second?.");
+
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateMethod)]
+        public void ClassificationInPreviewPane()
+        {
+            SetUpEditor(@"
+class Program
+{
+    int Main()
+    {
+        Foo$$();
+    }
+}");
+            this.InvokeCodeActionList();
+            var classifiedTokens = GetLightbulbPreviewClassification("Generate method 'Program.Foo'");
+            Assert.True(classifiedTokens.Any(c => c.Text == "void" && c.Classification == "keyword"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddImport)]
+        public void AddUsingExactMatchBeforeRenameTracking()
+        {
+            SetUpEditor(@"
+public class Program
+{
+    static void Main(string[] args)
+    {
+        P2$$ p;
+    }
+}
+
+public class P2 { }");
+
+            Editor.SendKeys(VirtualKey.Backspace, VirtualKey.Backspace, "Stream");
+
+            this.InvokeCodeActionList();
+            var expectedItems = new[]
+            {
+              "using System.IO;",
+              "System.IO.Stream",
+              "Generate class 'Stream' in new file",
+              "Generate class 'Stream'",
+              "Generate nested class 'Stream'",
+              "Generate new type...",
+              "Rename 'P2' to 'Stream'",
+              "Suppress CS0168",
+              "in Source"
+            };
```

Comment:
I think our time period of using stylecop has made me feel like close curly braces deserve their own blank line afterward. But I don't care that much.